src/sentry/api/endpoints/oauth_userinfo.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework.authentication import get_authorization_header
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.models import ApiToken, UserEmail
+
+
+@control_silo_endpoint
+class OAuthUserInfoEndpoint(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    def get(self, request: Request) -> Response:
+        try:
+            access_token = get_authorization_header(request).split()[1].decode("utf-8")
+        except IndexError:
+            return Response("No access token found.", status=401)
+        try:
+            token_details = ApiToken.objects.get(token=access_token)
+        except ApiToken.DoesNotExist:
+            return Response(status=400)
+
+        scopes = token_details.get_scopes()
+        if "openid" not in scopes:
+            return Response(status=403)
+
+        user = token_details.user
+        user_output = {"sub": user.id}
+        if "profile" in scopes:
+            profile_details = {
+                "name": user.name,
+                "avatar_type": user.avatar_type,
+                "avatar_url": user.avatar_url,
+                "date_joined": user.date_joined,
+            }
+            user_output.update(profile_details)
+        if "email" in scopes:
+            email = UserEmail.objects.get(user=user)
+            email_details = {"email": email.email, "email_verified": email.is_verified}
```

Comment:
The `email_verified` attribute is generally serialized as `is_verified`. While It is unfortunately a snake cased identifier we should try to use consistent names across various endpoints.

Reply:
I named this one `email_verified` since it's consistent with standard OIDC language (example from [Auth0](https://auth0.com/docs/get-started/apis/scopes/openid-connect-scopes)).

Also - this response contains more details on the user than just email stuff, if we just have `is_verified` do you think clients might be unsure as to what exactly is verified? 

On the other hand, maybe `is_verified`, by extension, implies that the user is verified, and not just the email, plus we would be consistent with _our_ codebase, rather than the OAuth standards, so there's a trade-off here.

I'd leave it this way but I'm good either way - lmk if you want me to change it over still and I will!

Reply:
Good points. Combined with the OIDC conventions discussed above, let's leave this as you have it.